vowpalwabbit/parse_args.cc

```diff
@@ -1643,16 +1662,32 @@ vw* initialize(std::string s, io_buf* model, bool skipModelLoad, trace_message_t
   return ret;
 }
 
-vw* initialize_escaped(
-    std::string const& s, io_buf* model, bool skipModelLoad, trace_message_t trace_listener, void* trace_context)
+vw* initialize_with_builder(int argc, char* argv[], io_buf* model, bool skipModelLoad, trace_message_t trace_listener,
+    void* trace_context, std::unique_ptr<VW::setup_base_i> learner_builder)
+{
+  std::unique_ptr<options_i, options_deleter_type> options(
+      new config::options_boost_po(argc, argv), [](VW::config::options_i* ptr) { delete ptr; });
```

Comment:
can we use `VW::make_unique` here instead of `new`?

Reply:
This is moved code, so we can address in separate PR. Although Jack confirmed that VW::make_unique right now doesn't have support for custom delete.